packages/discord.js/typings/index.d.ts

```diff
@@ -2632,19 +2626,28 @@ export interface MappedComponentTypes {
   [ComponentType.TextInput]: TextInputComponent;
 }
 
-export class Components extends null {
-  public static createComponent<T extends keyof MappedComponentTypes>(
-    data: APIMessageComponent & { type: T },
-  ): MappedComponentTypes[T];
-  public static createComponent<C extends Component>(data: C): C;
-  public static createComponent(data: APIMessageComponent | Component): Component;
-  public static createComponentBuilder<T extends keyof MappedComponentBuilderTypes>(
-    data: APIMessageComponent & { type: T },
-  ): MappedComponentBuilderTypes[T];
-  public static createComponentBuilder<C extends ComponentBuilder>(data: C): C;
-  public static createComponentBuilder(data: APIMessageComponent | ComponentBuilder): ComponentBuilder;
+export interface CreateChannelOptions {
+  allowFromUnknownGuild?: boolean;
+  fromInteraction?: boolean;
 }
 
+export function isThread(channel: AnyChannel | Channel): channel is ThreadChannel;
+export function isTextBased(channel: AnyChannel | Channel): channel is TextBasedChannel;
+export function isDMBased(channel: AnyChannel | Channel): channel is DMChannel | PartialDMChannel;
+export function isVoiceBased(channel: AnyChannel | Channel): channel is VoiceBasedChannel;
```

Comment:
I don't think we should have this union, and in fact, it should be just `AnyChannel`.

Reply:
I've tried this, but the TS complains that the type predicated isn't assignable to the parameter.